gui/default/syncthing/core/syncthingController.js

```diff
@@ -1623,17 +1646,14 @@ angular.module('syncthing.core')
 
         $scope.deviceFolders = function (deviceCfg) {
             var folders = [];
-            for (var folderID in $scope.folders) {
-                var devices = $scope.folders[folderID].devices;
-                for (var i = 0; i < devices.length; i++) {
-                    if (devices[i].deviceID === deviceCfg.deviceID) {
-                        folders.push(folderID);
+            $scope.folderList().forEach(function (folder) {
+                for (var i = 0; i < folder.devices.length; i++) {
+                    if (folder.devices[i].deviceID === deviceCfg.deviceID) {
+                        folders.push(folder.id);
                         break;
                     }
                 }
-            }
-
-            folders.sort(folderCompare);
+            });
```

Comment:
Why sort gone?

Reply:
Because it doesn't work: `folders` at this point is a list of IDs, not folderconfigs -> not sortable by label. The loop above is over `folderList()`, which is already sorted, so the result is sorted too.

Reply:
ah ok.